File: Programmers/Lv3/number_game.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> A, vector<int> B);

int main() {
    vector<int> A = {5, 1, 3, 7};
    vector<int> B = {2, 2, 6, 8};

    cout << solution(A, B);

    return 0;
}

int solution(vector<int> A, vector<int> B) {
    int answer = 0;

    sort(A.begin(), A.end());
    sort(B.begin(), B.end());

    if (A.front() > B.back()) {
        return 0;
    } else if (A.back() < B.front()) {
        return A.size();
    }

    int i = 0;
    int j = 0;
    for (auto a : A) {
        if (A[i] < B[j]) {
            answer++;
            i++;
        }
        j++;
    }

    return answer;
}

File: java/src/programmers/lv3/NumberGame.java
package programmers.lv3;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

public class NumberGame {
    public int solution(int[] A, int[] B) {
        int answer = 0;
        Arrays.sort(A);
        Arrays.sort(B);

        int j=0;
        for(int i =0; i < A.length; i++){
            for(; j < B.length;){
                if(A[i] < B[j++]){
                    answer++;
                    break;
                }
            }
        }
        return answer;
    }
    @Test
    void test(){
        Assertions.assertEquals(3, solution(new int[]{5,1,3,7}, new int[]{2,2,2,8}));
    }
}
